db/config.c

```diff
@@ -1330,10 +1320,7 @@ static int read_lrl_option(struct dbenv *dbenv, char *line,
             return -1;
 #endif
     } else if (tokcmp(tok, ltok, "legacy_defaults") == 0) {
-        /* NOOP: Already handled in pre_read_option(); check added here to keep
-         * handle_lrl_tunable() from throwing a warning as legacy_defaults has
-         * not been registered into the new tunables sub-system yet.
-         */
+        read_legacy_defaults(dbenv, options);
```

Comment:
"legacy_defaults" being a bit special, I'd add a line of comment explaining why this needs to be handled here (again).